webui/react/src/components/ModelRegistry.tsx

```diff
@@ -93,9 +105,30 @@ const ModelRegistry: React.FC<Props> = ({ workspace }: Props) => {
     activeSettings,
     isLoading: isLoadingSettings,
     settings,
-    updateSettings,
-    resetSettings,
-  } = useSettings<Settings>(settingConfig);
+    updateSettings: updateUseSettings,
+    resetSettings: resetUseSettings,
+  } = useSettings<ReducedSettings>(settingConfig);
+
+  const { params, updateParams } = useTypedParams(tableOffsetType, {
+    tableOffset: 0,
+  });
+
+  const resetSettings: ResetSettings = useCallback(
+    (reset) => {
+      if (reset?.includes('tableOffset')) updateParams({ tableOffset: 0 });
+      resetUseSettings(reset);
+    },
+    [resetUseSettings, updateParams],
+  );
```

Comment:
nit: given that we're only calling `resetSettings` in `resetFilters`, couldn't we just get rid of this by replacing `resetFilters` with

```ts
const resetFilters = useCallback(() => {
  resetUseSettings(filterKeys);
  updateParams({ tableOffset: 0 });
}, [resetUseSettings, updateParams]);
```